app/policies/user_policy.rb

```diff
@@ -0,0 +1,35 @@
+# app/policies/user_policy.rb
+
+class UserPolicy < ApplicationPolicy
+  attr_reader :current_user, :user
+
+  def initialize(current_user, user)
+    @current_user = current_user
+    @user = user
+  end
+
+  def show?
+    true
+  end
+
+  def feed?
+    user == current_user
+  end
+
+  def discover?
+    user == current_user
+  end
+  
+  def liked?
+    true
+  end
+
+  def view_private_profile_content?
+    Rails.logger.debug "Private: #{user.private?}, Current User: #{current_user.id}, User: #{user.id}, Followers: #{user.followers.map(&:id)}"
+    !user.private? || user == current_user || user.followers.include?(current_user)
+  end
+
+  def see_follow_request_button?
+    !user.private? || current_user != user || user.followers.include?(current_user)
```

Comment:
I think the boolean expression needs to be reevaluated here. If a user is private, I still want to be able to follow them. 

If I'm already following them, I want to be able to unfollow them. Is this handled by the same button?

Do we even need this?

Reply:
My fix:
```
def see_follow_request_button?
    current_user != user
  end
```

The only reason I included this is to hide the follow request button on the user's own profile. Please let me know if there is another, easier way.